packages/@guardian/source-react-components/src/button/styles.ts

```diff
@@ -29,51 +29,124 @@ const button = css`
 	text-decoration: none;
 	white-space: nowrap;
 
+	:disabled {
+		cursor: not-allowed;
+	}
+
 	&:focus {
 		${focusHalo};
 	}
 `;
 
+const transitionSpinnerBackground = css`
+	path,
+	circle {
+		transition: stroke ${transitions.medium};
+	}
+`;
+
 const primary = (
 	button: ButtonTheme = buttonThemeDefault.button,
+	isLoading = false,
 ): SerializedStyles => css`
 	background-color: ${button.backgroundPrimary};
 	color: ${button.textPrimary};
 
+	${isLoading &&
+	`circle {
+		stroke: ${button.backgroundPrimary};
+	}
+	path {
+		stroke: ${button.textPrimary}
+	}`}
```

Comment:
Can we extract all the `isLoading` styles into a separate block of CSS, and conditionally pass it to the `css` prop of the `button` element in `Button.tsx`?

```ts
export const isLoading = css`
    circle {
		stroke: ${button.backgroundPrimary};
	}
	path {
		stroke: ${button.textPrimary}
	}
    &:hover {
      circle {
			stroke: ${button.backgroundPrimaryHover};
		}
    }
   // ...
`
```

Reply:
big fan of this idea 👍